### ISSUE TITLE
Proposal: split out roles into individual repositories, with versioning, and published to Ansible Galaxy

### ISSUE THREAD

**Issue by adborden** [sovereign/sovereign#814]
Hey folks,

I'd like to propose that instead of having a single repo that contains all of sovereign, we split each role into its own repository with [semver](https://semver.org/) versioning and publishing to Ansible Galaxy. The main Sovereign repo would then become a [template repository](https://help.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-template-repository) that users would fork and customize. Users would then receive updates by updating the individual rules (`ansible-galaxy install`) rather than `git pull`.

What this will give us:
- Stability for users
- Faster integration of PRs
- Better customization

### Stability

Users can upgrade roles individually. When following [semver](https://semver.org/), major version bumps signal breaking changes so users don't need to :crossed_fingers: every time they `git pull`.

### Faster integration of PRs

Since each role contains only code for that role, there's fewer tests (manual or automated) that need to run when a change happens. This should make it easier to reivew and test Pull Requests.

### Better customization

If roles are published to Ansible Galaxy, the Sovereign roles become not much different than any other Ansible roles and it's pretty striaght forward to mix and match Sovereign roles with other Ansible roles.

**Comment by drybjed** [comment]
Why not convert the project to be an Ansible Collection instead? Individual Ansible roles will be phased out in the future when more and more features will be moved into Ansible Collections. These different methods of using Ansible are not really compatible with each other, so I would look into that direction instead.

**Comment by adborden** [comment]
Good point. In terms of this proposal, I think that would mean creating a separate repo for the sovereign collection and the main repo would still become a template repo containing getting started docs, requirements.yml, and the site.yml playbook. Does that sound right?

An Ansible collection would still be a monolithic repo, so we lose some of the benefits I'm after of having less to test when reviewing PRs (not that it's a bad thing).
